src/main/resources/api/service_openapi.yaml

```diff
@@ -754,6 +1071,27 @@ components:
           schema:
             $ref: '#/components/schemas/DataReferenceDescription'
 
+    DataRepoSnapshotReferenceResponse:
+      description: Response containing a reference to a Data Repo snapshot.
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/DataRepoSnapshotReference'
+
+    GoogleBucketReferenceResponse:
+      description: Response containing a reference to a Google bucket.
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/GoogleBucketReference'
+
+    BigQueryDatasetReferenceResponse:
+      description: Response containing a reference to a BiqQuery dataset.
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/BigQueryDatasetReference'
+
```

Comment:
this will also need updates based on #221  - adding description to `DataReferenceMetadata` ( I assume that's where it would go ), plus patch APIs for each reference type.